src/Core/View.js

```diff
@@ -458,4 +458,51 @@ View.prototype.execFrameRequesters = function execFrameRequesters(when, dt, upda
     }
 };
 
+const _eventCoords = new Vector2();
+/**
+ * Extract window coordinates from a mouse-event / touch-event
+ * @param {event} event - event can be a MouseEvent or a TouchEvent
+ * @param {number} touchIdx - finger index when using a TouchEvent (default: 0)
+ * @return {THREE.Vector2} - window coordinates (in pixels, 0-0 = top-left of the View)
+ */
+View.prototype.eventToWindowCoords = function eventToWindowCoords(event, touchIdx = 0) {
+    if (event.touches === undefined || !event.touches.length) {
+        return _eventCoords.set(event.offsetX, event.offsetY);
+    } else {
+        return _eventCoords.set(event.touches[touchIdx].pageX, event.touches[touchIdx].pageY);
```

Comment:
If I read correctly the doc on mdn, pageX and pageY are not equivalent to offsetX and Y in the case of touch events...

Reply:
pageX/Y are relative to the whole document (they are `window` coordinates in the web sense). I guess you need to take into account where viewerDiv is placed.

Reply:
I reused the existing code from `FirstPersonControls`.

Reply:
I replaced the pageXY by clientXY - it's closer to offsetXY.